scheduler/src/cook/mesos/constraints.clj

```diff
@@ -82,7 +79,15 @@
     [this _ vm-attributes _]
     (job-constraint-evaluate this _ vm-attributes)))
 
-(defrecord gpu-host-constraint [job]
+(defn build-novel-host-constraint
+  "Constructs a novel-host-constraint"
+  [job]
+  (let [previous-hosts (->> (:job/instance job)
+                            (remove #(true? (:instance/preempted? %)))
+                            (mapv :instance/hostname))]
+    (->novel-host-constraint job previous-hosts)))
+
+(defrecord gpu-host-constraint [job needs-gpus?]
```

Comment:
Yes, just updated the help message on build-gpu-host-constraint. It is meant to prevent gpu jobs from going on non-gpu hosts and non-gpu jobs from going on gpu hosts